ide/lsp.client/src/org/netbeans/modules/lsp/client/bindings/CompletionProviderImpl.java

```diff
@@ -226,36 +223,66 @@ public void render(Graphics grphcs, Font font, Color color, Color color1, int i,
+                                            } catch (InterruptedException | ExecutionException ex) {
+                                                Exceptions.printStackTrace(ex);
+                                                temp = i;
                                             }
-                                            @Override
-                                            public Action getGotoSourceAction() {
-                                                return null;
-                                            }
-                                        });
+                                            resolved = temp;
+                                        } else {
+                                            resolved = i;
+                                        }
+                                        if (resolved.getDocumentation() != null || resolved.getDetail() != null) {
+                                            resultSet.setDocumentation(new CompletionDocumentation() {
+                                                @Override
+                                                public String getText() {
+                                                    StringBuilder documentation = new StringBuilder();
+                                                    documentation.append("<html>\n");
+                                                    if (resolved.getDetail() != null) {
+                                                        documentation.append("<b>").append(escape(resolved.getDetail())).append("</b>");
+                                                        documentation.append("\n<p>");
+                                                    }
+                                                    if (resolved.getDocumentation() != null) {
+                                                        MarkupContent content;
+                                                        if (resolved.getDocumentation().isLeft()) {
+                                                            content = new MarkupContent();
+                                                            content.setKind("plaintext");
+                                                            content.setValue(resolved.getDocumentation().getLeft());
+                                                        } else {
+                                                            content = resolved.getDocumentation().getRight();
+                                                        }
+                                                        switch (content.getKind()) {
+                                                            case "markdown":
+                                                            default:
+                                                            case "plaintext": documentation.append("<pre>\n").append(content.getValue()).append("\n</pre>"); break;
```

Comment:
all cases pass here?

Reply:
@jlahoda Is this intentional?

Reply:
(I apologize for a late reply.) Yes, intentional for now. I have a an update for this, which actually converts the markdown to HTML (so that it can be shown nicely in the HTML pane), and also that converts the HTML to markdown for the server, but this involves a new external library, and it felt a little late in the release cycle to do that, so I opted for this simpler thing. The markdown should be still readable, and for the next release, it is really a few lines of changes here to do the conversion.

Reply:
No problem :) Thank you for your explanation.